lib/Target/ARM/ARMPGLTOptimizer.cpp

```diff
@@ -278,7 +279,7 @@ void ARMPGLTOpt::deleteOldCPEntries(SmallVectorImpl<int> &CPEntries) {
 
 bool ARMPGLTOpt::isSameBin(const GlobalValue *GV) {
   if (auto *F = dyn_cast<Function>(GV))
-    return MMI->getBin(F) == CurBin;
+    return (F->isRandPage() && F->getSectionPrefix() == CurBinPrefix);
```

Comment:
Can be simplified: `return F && F->isRandPage() && F->getSectionPrefix() == CurBinPrefix`